library/psa_crypto.c

```diff
@@ -1491,9 +1501,14 @@ static psa_status_t psa_finish_key_creation(
             status = psa_save_persistent_key( &attributes, buffer, length );
         }
 
-        if( buffer_size != 0 )
-            mbedtls_platform_zeroize( buffer, buffer_size );
-        mbedtls_free( buffer );
+#if defined(MBEDTLS_PSA_CRYPTO_SE_C)
+        if( driver == NULL )
```

Comment:
Looking at all of the hooks and checks for null drivers (here and in other functions) makes me wonder - why not introduce a table of pointers to functions (like in `cipher`) that would be populated on key registration/creation and then, instead of having checks in each function in runtime, a function dispatching mechanism would take care of it?
It probably would decrease complexity at the cost of code size though.

Reply:
I'd like a mechanism that doesn't require so much boilerplate on each call, but how to do it? Pretty much each method takes a different set of parameters. A variadic macro might help. But that's out of scope here, it's a job for https://github.com/ARMmbed/mbed-crypto/pull/174.